activestorage/lib/active_storage/service.rb

```diff
@@ -60,10 +61,18 @@ def configure(service_name, configurations)
       #
       # See MirrorService for an example.
       def build(configurator:, name:, service: nil, **service_config) # :nodoc:
+        if service_config.key?(:uri)
+          service_config = options_from_uri(**service_config)
+        end
+
         new(**service_config).tap do |service_instance|
           service_instance.name = name
         end
       end
+
+      def options_from_uri(uri:, **) # :nodoc:
```

Comment:
Finding it odd that this is taking an open set of other args that aren't documented.